modules/agar/src/test/ts/webdriver/RealMouseTest.ts

```diff
@@ -80,61 +80,73 @@ UnitTest.asynctest('RealMouseTest', (success, failure) => {
-
-  const other = SugarElement.fromTag('button');
-  Class.add(other, 'other');
-  Html.set(other, 'other-button');
-  Insert.append(container, other);
-
-  const normal = SugarElement.fromTag('button');
-  Html.set(normal, 'Normal');
-  Insert.append(container, normal);
-
-  Insert.append(SugarElement.fromDom(document.body), container);
-
-  const clickMe = SugarElement.fromTag('button');
-  Class.add(clickMe, 'click-me');
-  Html.set(clickMe, 'Click me!');
-  Insert.append(container, clickMe);
-
-  const count = Cell(0);
-  // add a MouseUp handler
-  const binder = DomEvent.bind(clickMe, 'mouseup', () => {
-    count.set(count.get() + 1);
+  let binder: EventUnbinder;
+  let container: SugarElement<HTMLDivElement>;
+  let count: Cell<number>;
+  let button: SugarElement<HTMLButtonElement>;
+  let other: SugarElement<HTMLButtonElement>;
+
+  before(() => {
+
+    const style = document.createElement('style');
+    style.innerHTML = 'button[data-test]:hover { background-color: blue; color: white; } button.other { background-color: blue; color: white; } button';
+    document.head.appendChild(style);
+
+    container = SugarElement.fromTag('div');
+    button = SugarElement.fromTag('button');
+    Attribute.set(button, 'data-test', 'true');
+    Html.set(button, 'hover-button');
+    Insert.append(container, button);
+
+    other = SugarElement.fromTag('button');
+    Class.add(other, 'other');
+    Html.set(other, 'other-button');
+    Insert.append(container, other);
+
+    const normal = SugarElement.fromTag('button');
+    Html.set(normal, 'Normal');
+    Insert.append(container, normal);
+
+    Insert.append(SugarElement.fromDom(document.body), container);
+
+    const clickMe = SugarElement.fromTag('button');
+    Class.add(clickMe, 'click-me');
+    Html.set(clickMe, 'Click me!');
+    Insert.append(container, clickMe);
+
+    count = Cell(0);
+    // add a MouseUp handler
+    binder = DomEvent.bind(clickMe, 'mouseup', () => {
+      count.set(count.get() + 1);
+    });
   });
 
-  try {
+  it('Should find buttons with same background color after hovering', async () => {
 
     await RealMouse.pMoveToOn('.other');
     await RealMouse.pMoveToOn('button[data-test]');
     const testButton = UiFinder.findIn(container, 'button[data-test]').getOrDie();
     Assert.eq('After hovering', Css.get(other, 'background-color'), Css.get(testButton, 'background-color'));
+  });
+
+  it('Should find click-me button, click on it and cleanup bedrock attribute', async () => {
     const clickMeTestButton = UiFinder.findIn(container, '.click-me').getOrDie();
     await RealMouse.pClick(clickMeTestButton);
     Assertions.assertEq('mouseup event has fired', 1, count.get());
     Assertions.assertEq(`button doesn't have ${RealMouse.BedrockIdAttribute} attribute`, false, Attribute.has(button, RealMouse.BedrockIdAttribute));
-  } finally {
+  });
+
+  after(() => {
     binder.unbind();
     Remove.remove(container);
-  }
-
+  });
```

Comment:
Our normal code style here is to put the `before` and `after` beside each other before any `it` calls.

Reply:
Done!